### PR TITLE
add initial version of PSA compliance test documents

### PR DESCRIPTION
**Pull request by NirSonnenschein**
PSA compliance test documents

### CODE REVIEW

**Comment by NirSonnenschein** [comment]
documents the addition of test from this PR: https://github.com/ARMmbed/mbed-os/pull/9312

**Comment by AnotherButler** [comment]
How does this fit into PR #967 ? Our typical porting guide template includes testing information at the bottom of the porting guide. Does this include testing information for all of PSA instead of just attestation? Is that it?

**Comment by AnotherButler** [comment]
We require engineering review for all new/changed content, so I'm tagging @avolinski for content review.
If he's busy or someone else would be a better fit for this review, please tag a different engineer (who didn't write the PR).

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+* `<test case number>` is:
+    * For crypto tests: `c001`, `c002`, and so on.
+    * For internal trusted storage tests: `s001`, `s002`, and so on.
+    * For attestation tests: `a001`, `a002`, and so on.
+
+#### Cryptographic configuration
+By default, the PSA compliance tests run using the default Mbed OS cryptography configuration (the configuration can be found at `features/mbedtls/inc/mbedtls/config.h` in the Mbed OS directory). When you use the default configuration, only tests supported by the configuration run.
+
+For an example of how to run the tests with a different configuration, see the [mbed-os-psa-compliance-tests-example repository](https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example).
```
Comment (GuyWi) [inline docs/porting/psa/testing_psa_service_compliance.md]: It is not immediately clear to me how https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example serves as an example of how to run the tests with a different configuration.

**Comment by AnotherButler** [comment]
@avolinski @alzix I talked with @NirSonnenschein in our docs reps meeting this morning, and I think this is in the right place. However, our [normal porting template](http://docs-previewer.herokuapp.com/documentation_docs/master/writing/mbed-os-vendor-partner-contributing-api-reference.html) usually contains more information. Does either of you know if there's a nontesting porting element to PSA compliance? If so, who will be providing this content?

**Comment by alekshex** [comment]
Hi @AnotherButler ,
psa compliance will have in addition an external example repository mainly for OOB.
later when mbedOS will be able to support multiple configurations per build, this repository will become obsolete
https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example

i am not sure if we want to mention it in the documentation
i am not able to access the porting template link in your message it result in 'not found' page
http://docs-previewer.herokuapp.com/documentation_docs/master/writing/mbed-os-vendor-partner-contributing-api-reference.html

**Comment by AnotherButler** [comment]
I should have mentioned that link only works on the Arm network.

Do you and @ndevillard think a porting guide, if relevant, should wait until Mbed OS can support multiple configurations?

**Comment by alekshex** [comment]
Hi @AnotherButler ,
i am on the Arm network, still...
Not Found
The requested resource was not found on this server.

**Comment by AnotherButler** [comment]
Interesting. That works for me. I'll have to look into it. Here's the GH version: https://github.com/ARMmbed/documentation_docs/blob/master/Docs/Templates/mbed-os-port-template.md

**Comment by AnotherButler** [comment]
@avolinski or @orenc17 Could one of you please approve this PR if it's ready to merge? We require approval from an engineer who didn't create the PR before we can merge content changes/additions. (Or feel free to tag a different engineer for approval if you're not the right person.)

**Comment by GuyWi** [comment]
@avolinski  or @orenc17  , please review and approve this PR.
Thank you!

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<h2 id="crypto-port">Testing PSA service compliance</h2>
+
+Mbed OS provides an integrated version of the Platform Security Architecture (PSA) test suite running over the Greentea test infrastructure. You can use these tests on PSA services that you develop or customize on PSA-enabled platforms to verify the compliance of the services to the PSA specification.
+
+For information about PSA compliance tests, see [the PSA compliance test repository](https://github.com/ARM-software/psa-arch-tests).
+
+### Test directory layout
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:7]: ### Tests directory layout

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+### Compiling and running tests
+
+To compile and run PSA compliance tests in Mbed OS, run the following command:
+
+```
+mbed test -t <toolchain> -m <target> -n mbed-os-components-target_psa-tests-compliance_<test suite>_<test case number>
+```
+Where:
+
+* `<toolchain>` may be `GCC_ARM`, `ARM` or `IAR`.
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: `ARM`,  `ARMC6`, `GCC_ARM` or  `IAR`

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+* `<target>` is  your target.
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: `<target>` is  your target platform

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+* Initial attestation test suite (`components/TARGET_PSA/TESTS/compliance_attestation` in the Mbed OS directory).
+
+Every test suite contains a folder for each of the test scenarios. Each test scenario may run one or more test vectors.
+
+### Compiling and running tests
+
+To compile and run PSA compliance tests in Mbed OS, run the following command:
+
+```
+mbed test -t <toolchain> -m <target> -n mbed-os-components-target_psa-tests-compliance_<test suite>_<test case number>
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: ... -n components-target_psa-tests-compliance_......
without mbed-os- prefix

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+* `<test suite>` is:
+    * For crypto tests: `crypto-test`.
+    * For internal trusted storage tests: `its-test`.
+    * For attestation tests: `attestation-test`.
+
+* `<test case number>` is:
+    * For crypto tests: `c001`, `c002`, and so on.
+    * For internal trusted storage tests: `s001`, `s002`, and so on.
+    * For attestation tests: `a001`, `a002`, and so on.
+
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:39]: Wildcards can be used to compile and run the entire set of PSA compliance tests:
`mbed test -t <toolchain> -m <target> -n components-target_psa-tests-compliance_*`

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 4 above)
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: (Please note, the tests can run only on PSA enabled target platforms)

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+### Cryptographic configuration
+By default, the PSA compliance tests run using the default Mbed OS cryptography configuration (the configuration can be found at `features/mbedtls/inc/mbedtls/config.h` in the Mbed OS directory). When you use the default configuration, only tests supported by the configuration run.
+
+You can override the default Mbed OS cryptography configuration using an `mbed_app.json` file. For an example of how to use an `mbed_app.json` file to override the default cryptography configuration, see the [mbed-os-psa-compliance-tests-example repository](https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example).
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: You can override the default Mbed OS cryptography configuration using a `mbed_app.json` file.
Or supplying your custom configuration file directly as command line compilation flag.

**Review by alekshex — CHANGES_REQUESTED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 6 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: @avolinski which wildcards can you use? Or do you always use an asterisk (*)?

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 4 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski, do we have a list of PSA-compliant platforms?

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 7 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski , can we specify what the flag is?

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 5 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Fixed.

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 3 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Fixed.

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 4 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Modified.

Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Modified.

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 7 above)
Comment (alekshex) [reply docs/porting/psa/testing_psa_service_compliance.md]: Or supplying your custom configuration file directly as command line parameter flag MBEDTLS_USER_CONFIG_FILE=<custom config file path>
By passing it to the mbed compilation command

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 6 above)
Comment (NirSonnenschein) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: * is the only relevant wildcard we are aware of in this case.

Comment (NirSonnenschein) [reply docs/porting/psa/testing_psa_service_compliance.md:39]: "*" github renders it as a bullet

**Comment by NirSonnenschein** [comment]
@GuyWi thanks the latest exit looks fine to me.
@avolinski thanks for the notes. can you approve the fixes by Guy or do you want anything else?

**Review by AnotherButler — APPROVED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 7 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: Updated.

**Comment by GuyWi** [comment]
Thanks, @NirSonnenschein !
@avolinski, please approve.

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+### Cryptographic configuration
+
+By default, the PSA compliance tests run using the default Mbed OS cryptography configuration. (You can find the configuration at `features/mbedtls/inc/mbedtls/config.h` in the Mbed OS directory). When you use the default configuration, only tests supported by the configuration run.
+
+You can override the default Mbed OS cryptography configuration using an `mbed_app.json` file, or by passing the `MBEDTLS_USER_CONFIG_FILE=<path to your custom configuration file>` flag in the compilation command.
+
+For an example of how to use an `mbed_app.json` file to override the default cryptography configuration, please see the [mbed-os-psa-compliance-tests-example repository](https://github.com/ARMmbed/mbed-os-psa-compliance-tests-example).
+
+### Sample test output
+
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md:55]: When the tests run, they output state information to the serial output. Below is an example of a successful run:

mbedgt: mbed-host-test-runner: started
mbedgt: checking for GCOV data...
mbedgt: test on hardware with target id: 190000006c0f1507036c0f1500000000000000002e127069
mbedgt: test suite 'components-target_psa-tests-compliance_its-test_s009' ............................ OK in 55.01 sec
        test case: 'Check1' .......................................................................... OK in 0.40 sec
mbedgt: mbed-host-test-runner: started
mbedgt: checking for GCOV data...
mbedgt: test on hardware with target id: 190000006c0f1507036c0f1500000000000000002e127069
mbedgt: test suite 'components-target_psa-tests-compliance_its-test_s008' ............................ OK in 55.21 sec
        test case: 'Check1' .......................................................................... OK in 0.10 sec
        test case: 'Check2' .......................................................................... OK in 0.09 sec

mbedgt: test suite report:
| target                          | platform_name       | test suite                                           | result  | elapsed_time (sec) | copy_method |
|---------------------------------|---------------------|------------------------------------------------------|---------|--------------------|-------------|
| CY8CKIT_062_WIFI_BT_PSA-GCC_ARM | CY8CKIT_062_WIFI_BT | components-target_psa-tests-compliance_its-test_s001 | OK      | 19.45              | default     |
| CY8CKIT_062_WIFI_BT_PSA-GCC_ARM | CY8CKIT_062_WIFI_BT | components-target_psa-tests-compliance_its-test_s002 | OK      | 55.34              | default     |

File: docs/porting/psa/testing_psa_service_compliance.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+When the tests run, they output state information to the serial output. Below is an example of a successful run (running the first crypto test scenario):
```
Comment (alekshex) [inline docs/porting/psa/testing_psa_service_compliance.md]: please remove this section and below and replace with one above. Thank you!

**Review by alekshex — CHANGES_REQUESTED** [review]

File: docs/porting/psa/testing_psa_service_compliance.md (hunk 9 above)
Comment (GuyWi) [reply docs/porting/psa/testing_psa_service_compliance.md]: @avolinski, updated.
Please review and approve.
Thanks!

**Review by alekshex — APPROVED** [review]

**Pull request merged**